tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts

```diff
@@ -343,6 +336,54 @@ export class ScalarCardContainer implements CardRenderer, OnInit {
       startWith([] as ScalarCardDataSeries[])
     );
 
+    this.selectedTime$ = combineLatest([
+      partitionedSeries$,
+      this.store.select(getMetricsSelectedTime),
+      this.store.select(getMetricsXAxisType),
+    ]).pipe(
+      map(([series, selectedTime, xAxisType]) => {
+        if (xAxisType !== XAxisType.STEP || !selectedTime) return null;
+
+        let minStep = Infinity;
+        let maxStep = -Infinity;
+        for (const {points} of series) {
+          for (const point of points) {
+            minStep = minStep > point.x ? point.x : minStep;
+            maxStep = maxStep < point.x ? point.x : maxStep;
+          }
+        }
+
+        if (
+          // Case when selectedTime contains extents.
+          (selectedTime.start.step <= minStep &&
+            selectedTime.end &&
+            maxStep <= selectedTime.end.step) ||
+          // Case when start of selectedTime is within extent.
+          (minStep <= selectedTime.start.step &&
+            selectedTime.start.step <= maxStep) ||
+          // Case when end of selectedTime is within extent.
+          (selectedTime.end &&
+            minStep <= selectedTime.end?.step &&
+            selectedTime.end?.step <= maxStep)
+        ) {
+          return {...selectedTime, clipped: false};
+        }
+
+        if (maxStep <= selectedTime.start.step) {
```

Comment:
Would be nice to continue with the comments here and in the final case.
Is this "Case when selectedTime is entirely greater than extents"?